qiskit/test/base.py

```diff
@@ -33,64 +32,47 @@
 
 try:
     import fixtures
-    from testtools.compat import advance_iterator
-    from testtools import content
+    import testtools
 
     HAS_FIXTURES = True
 except ImportError:
     HAS_FIXTURES = False
 
-from qiskit.exceptions import MissingOptionalLibraryError
 from .decorators import enforce_subclasses_call
-from .runtest import RunTest, MultipleExceptions
 from .utils import Path, setup_test_logging
 
 
 __unittest = True  # Allows shorter stack trace for .assertDictAlmostEqual
 
 
-def _copy_content(content_object):
-    """Make a copy of the given content object.
+# If testtools is installed use that as a (mostly) drop in replacement for
+# unittest's TestCase. This will enable the fixtures used for capturing stdout
+# stderr, and pylogging to attach the output to stestr's result stream.
+if HAS_FIXTURES:
 
-    The content within ``content_object`` is iterated and saved. This is
-    useful when the source of the content is volatile, a log file in a
-    temporary directory for example.
+    class BaseTestCase(testtools.TestCase):
+        """Base test class."""
 
-    Args:
-    content_object (content.Content): A ``content.Content`` instance.
+        # testtools maintains their own version of assert functions which mostly
+        # behave as value adds to the std unittest assertion methods. However,
+        # for assertEquals and assertRaises modern unittest has diverged from
+        # the forks in testtools and offer more (or different) options that are
+        # incompatible testtools versions. Just use the stdlib versions so that
+        # our tests work as expected.
+        assertRaises = unittest.TestCase.assertRaises
+        assertEqual = unittest.TestCase.assertEqual
```

Comment:
Do we need to account for all the other related `assertEqual` methods (`assertListEqual`, `assertDictEqual` etc), which `unittest` calls out to, and `assertRaisesRegex`/`assertWarns`/`assertWarnsRegex` as well?

Reply:
These were the 2 cases I hit in running the test suite locally. They caused a hard error, testtools's `assertEqual()` didn't have a `msg` arg and `assertRaises()` didn't work as a context manager. For the other cases I'm guess either testtools is just missing them or are compatible with the current versions of python's stdlib

Reply:
Sounds good to me, then - if more things pop up we can always squash them then.

Reply:
`testtools.TestCase` does define other assertions with the same deviation (that they take an argument `message=''` instead of `msg=None`) from `unittest.TestCase`: `assertIn()` `assertIsNone()` `assertIsNotNone()` `assertIs()` `assertIsNot()` `assertNotIn()`.

Strangely `testtools.TestCase.assertIsinstance()` breaks the pattern and uses `msg=None` just like for `unittest`.

While the qiskit test suite does use I think all of those assertions, I guess we never use the `msg` arg, or only use it as a positional argument.